src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -554,3 +554,57 @@ export function updateGraphObjectWithNewSharedModelIds(
     }
   }
 }
+
+// This is a modified version of CODAP V2's SvgScene.pathBasis which was extracted from protovis
+export const pathBasis = (p0: Point, p1: Point, p2: Point, p3: Point) => {
+  /**
+   * Matrix to transform basis (b-spline) control points to bezier control
+   * points. Derived from FvD 11.2.8.
+   */
+  const basis = [
+    [ 1/6, 2/3, 1/6,   0 ],
+    [   0, 2/3, 1/3,   0 ],
+    [   0, 1/3, 2/3,   0 ],
+    [   0, 1/6, 2/3, 1/6 ]
+  ];
+
+  /**
+   * Returns the point that is the weighted sum of the specified control points,
+   * using the specified weights. This method requires that there are four
+   * weights and four control points.
+   */
+  const weight = (w: number[]) => {
+    return {
+      x: w[0] * p0.x + w[1] * p1.x + w[2] * p2.x + w[3] * p3.x,
+      y: w[0] * p0.y  + w[1] * p1.y  + w[2] * p2.y  + w[3] * p3.y
+    };
+  };
+
+  const b1 = weight(basis[1]);
+  const b2 = weight(basis[2]);
+  const b3 = weight(basis[3]);
+
+  return `C${b1.x},${b1.y},${b2.x},${b2.y},${b3.x},${b3.y}`;
```

Comment:
I think these should be rounded off - they are in pixel units I think?  So it's not really necessary to have so many decimals: 

`<path ... d="M0,-1148.5238095238096,C0,-1148.5238095238096,0,-1148.5238095238096,0.16666666666666666,-1147.9601756388208C0.3333333333333333,-1147.396541753832,0.6666666666666666,...`